### DISCUSSION TITLE
Best way to redirect to `www` with Coolify?

### DISCUSSION THREAD

**Discussion by Christopher-Hayes** [coollabsio/coolify#1999] [Q&A] [answered]
I've always set up my sites to redirect the base domain to `www.` (LMK if that's that no longer considered necessary).

While setting that redirect up, I was having trouble getting it working in Coolify. If you use Vercel or Netlify, you're used to the platform "automagically" setting that up for you.

**My setup** - Hetzner server behind Cloudflare CDN proxy. I have SSL/TLS set to "full" in Cloudflare. It hosts a static site with Nixpacks.

**The first thing I tried** was setting up a CNAME record for `www` to go to my base domain. However, the `www` site 404s if I do that. I do have a wildcard `*` A record for easily setting up projects on subdomains, not sure if that's conflicting.

**The second thing I tried** was a Traefik config to have its dynamic router redirect to www. I tried that at both the server proxy config level and the project level without luck, but I probably just not using the right config values.

**The last thing I tried which _sort of_ works** - I have the project assigned to BOTH my root domain and the `www` subdomain. At least the site does not 404. I'm not sure this is correct though because the root domain does not redirect to `www`, it just shows the site in both places, sort of like it's 1 site in 2 places.

**Comment by actraiser** [comment, marked as answer]
Since you are using Cloudflare you can just select your domain on their Dashboard and chose Rules => Redirect Rules on the left Menu. There you can easily build a rule for this. No need to involve additional DNS-Settings or Traeffik.

Greets
-act

**Reply by Christopher-Hayes** [reply]
Good point, thanks!

**Edit:** It's working.

For future users - this is the setup I used. It's a "dynamic redirect" because I want it to work on all paths to my base domain. An "expression" is used so that the destination has the same path. "Preserve query string" checked to keep that as well.

Expression: `concat("https://www.hayes.software", http.request.uri.path)`

![image](https://github.com/coollabsio/coolify/assets/6013871/67c0fd0d-3841-4ac5-bb53-e9401286d0ca)

**Reply by enesbala5** [reply]
This doesn't seem to work for me. However I managed to get it working by adding a CNAME key:

Type: CNAME
Name: @ (this would be the domain without the www - value will be retrieved by Cloudflare automatically, no need to type it)
Value: www.domain.com

I have also added the Dynamic Redirect in case it's doing something I am not able to notice - in any case this change made it work now.

Edit: I checked whether it would work without the Page Redirect rule and it doesn't. So thank you @Christopher-Hayes and @actraiser . The CNAME was the missing piece for me - hope it's useful for others too.

**Comment by stonediggity** [comment]
Thanks for this I've been pulling my hair out trying to get this sorted and the cloudflare dynamic redirect rule worked a treat for me!
